src/e2e/puppeteer/helpers/waitForTransitionEnd.ts

```diff
@@ -0,0 +1,18 @@
+import { page } from '../setup'
+
+/** Wait for a transition to finish. */
+async function waitForTransitionEnd(selector: string) {
+  await page.evaluate(selector => {
+    return new Promise<void>(resolve => {
+      const transition = document.querySelector(selector)
+      if (!transition) throw new Error(`'${selector}' selector not found`)
+      const onEnd = function () {
```

Comment:
```suggestion
      const onEnd = () => {
```

I would prefer to see this as an arrow function for consistency. This project uses arrow functions almost everyone, and actual `function` blocks have some odd quirks/footguns.